nodejs/eks/cluster.ts

```diff
@@ -483,14 +485,55 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
             return mappingYaml;
         });
     }
-    const eksNodeAccess = new k8s.core.v1.ConfigMap(`${name}-nodeAccess`, {
-        apiVersion: "v1",
-        metadata: {
-            name: `aws-auth`,
-            namespace: "kube-system",
-        },
-        data: nodeAccessData,
-    }, { parent: parent, provider: provider });
+    /*
+    * We must kubectl apply the aws-auth configmap since the creation of an
+    * AWS managed node group automatically triggers AWS to create the aws-auth
+    * configmap for us.
+    *
+    * For managed nodegroups:
+    *   To properly create aws-auth, we need to kubectl apply the configmap
+    *   instead of kubectl create it, if not it will fail with an existing
+    *   resource error message.
+    *
+    * For self-managed nodegroups:
+    *   We continue to always create the aws-auth configmap.
+    *
+    * As a work around, we'll kubectl apply to satisfy both node group cases.
+    *
+    * TODO(metral)
+    * When https://git.io/JeXom is resolved, we can move away from kubectl
+    * shell out.
+    */
+    pulumi.all([
+        kubeconfig,
+        nodeAccessData,
+    ]).apply(([kc, accessData]) => {
+        const tmpKubeconfig = tmp.fileSync();
+        const tmpJson = tmp.fileSync();
+
+        // Dump the kubeconfig to a file.
+        fs.writeFileSync(tmpKubeconfig.fd, JSON.stringify(kc));
+
+        // Compute the required aws-auth JSON manifest and dump it to a file.
+        fs.writeFileSync(tmpJson.fd, `{
+"apiVersion": "v1",
+"kind": "ConfigMap",
+"metadata": {
+  "name": "aws-auth",
+  "namespace": "kube-system",
+  "labels": {
+    "app.kubernetes.io/managed-by": "pulumi"
+  }
+},
+"data": ${JSON.stringify(accessData)}
+}`);
+
+        // Call kubectl to apply the aws-auth manifest.
+        childProcess.execSync(`kubectl apply -f ${tmpJson.name}`, {
+            stdio: "ignore",
+            env: { ...process.env, "KUBECONFIG": tmpKubeconfig.name },
+        });
+    });
```

Comment:
Minor thing - since we are now doing this in two places - should we extract the common part out into a single function so we can maintain in one place?

Reply:
Two places? We're only doing this in `createCore()`. What other location are you referring to?

IIUC, we also do `kubectl apply` for VPC if that's what you're referring to, but that is a separate use case from this configmap as we allow user input to be accepted for the VPC.

Reply:
We do almost the exact same `kubectl apply` in VPC CNI.  Sharing the tmp file creation/writing, processing spawning, etc. in one place seems like it would help us ensure that these are maintained.